### ISSUE TITLE
Support for OAuth before ClientLogin is shut down 2015-04-20

### ISSUE THREAD

**Issue by jeremywadsack** [eyecatchup/php-webmaster-tools-downloads#19]
With the [impending April 2015 shutdown of ClientLogin](http://googledevelopers.blogspot.ch/2015/02/reminder-clientlogin-shutdown-scheduled.html) support at Google, has anyone investigated adding OAuth support to this library?

**Comment by clinttepe** [comment]
I agree this really needs to happen! I will be implementing OAuth on another project soon and am willing to help.

**Comment by jeremywadsack** [comment]
I found [this library](https://github.com/adoy/PHP-OAuth2) but as I don't understand OAuth2 it's probably faster for someone else to do this. Although I'm going to have to learn it to adapt a bunch of other feeds.

**Comment by clinttepe** [comment]
Great find, thank you!

**Comment by jeremywadsack** [comment]
FYI, John Mueller [says that `ClientLogin` will remain for the CSV downloads](http://googlewebmastercentral.blogspot.com/2015/03/deprecation-of-old-webmaster-tools-api.html) associated with [Google's Python library similar to this](https://code.google.com/p/webmaster-tools-downloads/). I don't know if it will be available for other downloads.

There's also suggestion from [this StackOverflow post](http://stackoverflow.com/a/16308531/201911) that the downloads may not work with OAuth2.

**Comment by gbusoni** [comment]
I was using this class with some small customizations, has been working fine form months and now is not working anymore, I get 403 error on login.

**Comment by gbusoni** [comment]
http://googlewebmastercentral.blogspot.it/2015/03/deprecation-of-old-webmaster-tools-api.html

**Comment by jeremywadsack** [comment]
@gbusoni We have had '403' errors for ClientLogin off-and-on for a while. They are generally related to frequency with which we hit the API and giving it a rest usually reduces the errors. Changing to 2-Step Authentication reduced this error count significantly. See #4.

We are not getting excessive 403 errors today for TOP_QUERIES or TOP_URLS (of ~35k requests we saw about 230 failures), so I don't think `ClientLogin` has been deprecated for those. I just tested this for a single site for CONTENT_KEYWORDS, CONTENT_ERRORS, INTERNAL_LINKS, and EXTERNAL_LINKS and none of those returned a 403 for me. 

The target date for take-down is Monday and it's possible these latter reports will be disabled, but the TOP_QUERIES and TOP_URLS should continue to be available, based on what I understood John Mueller to be saying. That said, I'll be watching this closely on Monday.

**Comment by jeremywadsack** [comment]
As a follow up after the shut-down date, we are still able to get TOP_QUERIES, TOP_URLS, CONTENT_KEYWORDS, CONTENT_ERRORS, INTERNAL_LINKS, and EXTERNAL_LINKS without issue. So the `ClientLogin` appears to work for all of this for the time being using this code.

I have not tried using this to get a list of sites or crawl errors as we switched over to Webmaster API v3 for that.

**Comment by gbusoni** [comment]
I started to get authentication refused. In the past, after some troubleshooting for a similar problem (was refusing authentication becauseof old browser), was working (till january i used it everyday). Then I didn't use it for couple of months and now is not working again. the problem is that I don't remember how I fixed this issue last time. You have any clue?

G

---

Da: Jeremy Wadsack [notifications@github.com]
Inviato: martedì 21 aprile 2015 22.42
A: eyecatchup/php-webmaster-tools-downloads
Cc: Giorgio Busoni
Oggetto: Re: [php-webmaster-tools-downloads] Support for OAuth before ClientLogin is shut down 2015-04-20 (#19)

As a follow up after the shut-down date, we are still able to get TOP_QUERIES, TOP_URLS, CONTENT_KEYWORDS, CONTENT_ERRORS, INTERNAL_LINKS, and EXTERNAL_LINKS without issue. So the ClientLogin appears to work for all of this for the time being using this code.

I have not tried using this to get a list of sites or crawl errors as we switched over to Webmaster API v3 for that.

**Comment by btibor91** [comment]
I think that this time is `ClientLogin` gone forever - http://googlewebmastercentral.blogspot.com/2015/08/with-new-search-analytics-api-its-now.html

**Comment by jeremywadsack** [comment]
Confirmed. We have authentication refused for this code as of today and are removing it from our tools. While the new, official Google API only provides queries, pages, and crawl errors, that's our primary data source anyway.
